tests/AulaTests.hs

```diff
@@ -163,5 +166,11 @@ boundary mn mx = frequency
     , (98, choose (mn, mx))
     ]
 
+nonEmptyST :: Gen ST
+nonEmptyST = (cs :: String -> ST) . getNonEmpty <$> arb
+
 nonEmptyMarkdown :: Gen Document
-nonEmptyMarkdown = Markdown . (cs :: String -> ST) . getNonEmpty <$> arb
+nonEmptyMarkdown = Markdown <$> nonEmptyST
+
+validTopicDescription :: Gen Description
+validTopicDescription = Description . Text.take Constant.topicDescMaxLength <$> nonEmptyST
```

Comment:
summarizing a brief off-line discussion: removing noise from the arbitrary instances to accomodate application logic is problematic because it introduces assumptions that are hard to look up.  worse, if different parts of the application logic have overlapping domains, we will be tempted to generate the intersection, punching holes in our data that are hard to track.

however:

1. we already do if (accomodate app logic) in `src/Arbitrary.hs`, and it's confusing to sometimes do sometimes have the nice-noise generators there and sometimes in other places like this one
2. white noise is mostly useless

in the long run, it would be nice to have more sophisticated testing (andorp: feel free to write a few lines on your ideas for the record :-).  for now, i will try to find the time soon to move these happy-noise generators into `src/Arbitrary.hs` and copy this reasoning to the module haddock.